backend/src/schema/resolvers/todos.js

```diff
@@ -0,0 +1,50 @@
+const { decodedToken } = require("../../utils/decode");
+
+const { todoListData } = require("../../data");
+
+module.exports.TodoResolver = {
+  Todo: {
+    author(todo) {
+      return todoListData.find(value => {
+        return todo.id === value.id;
+      }).author;
+    }
+  },
+  Query: {
+    todos: (object, params, context) => {
+      const decoded = decodedToken(context.token);
+      return todoListData.filter(todo => todo.author.name === decoded.username);
+    }
+  },
+  Mutation: {
+    delToDo: (object, params, context) => {
+      const decoded = decodedToken(context.token);
+      let index = todoListData.findIndex(todoData => todoData.id === params.id);
+      todoListData.splice(index, 1);
+      return todoListData.filter(todo => todo.author.name === decoded.username);
+    },
+    addToDo: (object, params, context) => {
+      const decoded = decodedToken(context.token);
+      var maxid = 0;
+      todoListData.map(obj => {
```

Comment:
I wouldn't call it "not proper", but reduce works better. I usually work with Kotlin which has much better Lambda implementations and some names are different. 😅

Could you explain the three dots in front of a map? ```...todoListData```

Reply:
`map` is always returning an array and the callback should return a value. So you're using a `map` instead of a `forEach`. `map` is supposed to *map* an array into another array.

The `...` notation is the destructuring assignment. Or call it "unpacking". https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
So my suggestion is to map the todoListData to an array of ids and call `Math.max` on the ids. The `0` is a placeholder for the edge case of an empty array.

Reply:
hm ok thank you